engine/src/build/toolchain/apple/swiftc.py

```diff
@@ -0,0 +1,649 @@
+
+  swift_toolchain_path = args.swift_toolchain_path
+  if not swift_toolchain_path:
+    swift_toolchain_path = os.path.join(os.path.dirname(args.sdk_path),
+                                        'XcodeDefault.xctoolchain')
+    if not os.path.isdir(swift_toolchain_path):
+      swift_toolchain_path = ''
+
+  command = [f'{swift_toolchain_path}/usr/bin/swiftc'] + swiftc_args
+  if extras_args:
+    command.extend(extras_args)
+
+  process = subprocess.Popen(command)
+  process.communicate()
+
+  if process.returncode:
+    sys.exit(process.returncode)
+
+  if args.fix_generated_header:
+    fix_generated_header(header_path,
+                         args.header_path,
+                         src_dir=os.path.abspath(args.src_dir) + os.path.sep,
+                         gen_dir=os.path.abspath(args.gen_dir) + os.path.sep)
+
+
+def generate_depfile(args, output_file_map):
+  """Generates compilation depfile according to `args`.
+
+  Parses all intermediate depfile generated by the Swift compiler and
+  replaces absolute path by relative paths (since ninja compares paths
+  as strings and does not resolve relative paths to absolute).
+
+  Converts path to the SDK and toolchain files to the sdk/xcode_link
+  symlinks if possible and available.
+  """
+  xcode_paths = {}
+  if os.path.islink(args.sdk_path):
+    xcode_links = os.path.dirname(args.sdk_path)
+    for link_name in os.listdir(xcode_links):
+      link_path = os.path.join(xcode_links, link_name)
+      if os.path.islink(link_path):
+        xcode_paths[os.path.realpath(link_path) + os.sep] = link_path + os.sep
+
+  out_dir = os.getcwd() + os.path.sep
+  src_dir = os.path.abspath(args.src_dir) + os.path.sep
+
+  depfile_content = collections.defaultdict(set)
+  for value in output_file_map.values():
+    partial_depfile_path = value.get('dependencies', None)
+    if partial_depfile_path:
+      with open(partial_depfile_path, encoding='utf8') as stream:
+        for line in stream:
+          output, inputs = line.split(' : ', 2)
+          output = os.path.relpath(output, out_dir)
+
+          # The depfile format uses '\' to quote space in filename. Split the
+          # list of file while respecting this convention.
+          for path in re.split(r'(?<!\\) ', inputs):
+            for xcode_path in xcode_paths:
+              if path.startswith(xcode_path):
+                path = xcode_paths[xcode_path] + path[len(xcode_path):]
+            if path.startswith(src_dir) or path.startswith(out_dir):
+              path = os.path.relpath(path, out_dir)
+            depfile_content[output].add(path)
+
+  with FileWriter(args.depfile_path) as stream:
+    for output, inputs in sorted(depfile_content.items()):
+      stream.write(f'{output}: {" ".join(sorted(inputs))}\n')
+
+
+def compile_module(args, extras_args, build_signature):
+  """Compiles Swift module according to `args`."""
+  for path in (args.target_out_dir, os.path.dirname(args.header_path)):
+    ensure_directory(path)
```

Comment:
Why would you make the make the header_path a directory?

Reply:
This is checking `dirname(header_path)` (which is a directory) to ensure the output directory into which we write the header (`gen/flutter/shell/platform/darwin/ios/$target_name` in our case) exists.